py-polars/polars/datatypes/convert.py

```diff
@@ -373,6 +373,46 @@ def dtype_to_py_type(dtype: PolarsDataType) -> PythonDataType:
         raise NotImplementedError(msg) from None
 
 
+def _dtype_to_init_repr_list(dtype: List, prefix: str) -> str:
+    class_name = dtype.__class__.__name__
+    inner_repr = dtype_to_init_repr(dtype.inner, prefix)
+    init_repr = f"{prefix}{class_name}({inner_repr})"
+    return init_repr
+
+
+def _dtype_to_init_repr_array(dtype: Array, prefix: str) -> str:
+    class_name = dtype.__class__.__name__
+    inner_repr = dtype_to_init_repr(dtype.inner, prefix)
+    init_repr = f"{prefix}{class_name}({inner_repr}, width={dtype.width})"
+    return init_repr
+
+
+def _dtype_to_init_repr_struct(dtype: Struct, prefix: str) -> str:
+    class_name = dtype.__class__.__name__
+    inner_list = [
+        f'"{field_name}": {dtype_to_init_repr(inner_dtype, prefix)}'
+        for field_name, inner_dtype in dict(dtype).items()
+    ]
+    inner_repr = "{" + ", ".join(inner_list) + "}"
+    init_repr = f"{prefix}{class_name}({inner_repr})"
+    return init_repr
+
+
+def dtype_to_init_repr(dtype: PolarsDataType | None, prefix: str) -> str:
```

Comment:
This should not accept `None` as input. Calling this on a `None` value makes no sense.

I guess we can set `prefix: str = "pl."` as a default.

Reply:
Moved the None handling to the Array/List versions (could not totally remove it as the inner dtype is annotated as possible None)